### PR TITLE
http: empty reply connection are not left intact

### PR DESCRIPTION
**Pull request by bagder**
... so mark the connection as closed in this condition to prevent that
verbose message to wrongly appear.

Reported-by: Matt Holt
Bug: https://twitter.com/mholt6/status/1352130240265375744

### CODE REVIEW

File: lib/http.c [hunk 1]
```diff
@@ -1596,6 +1596,8 @@ CURLcode Curl_http_done(struct Curl_easy *data,
        read from the HTTP server (that counts), this can't be right so we
        return an error here */
     failf(data, "Empty reply from server");
+    /* Mark it as closed to avoid the "left intact" message */
+    connclose(conn, "Empty reply from server");
```
Comment (jay) [inline lib/http.c:1600]: At this point is it possible that an http2 stream may be bad but the connection still good?

Comment (bagder) [reply lib/http.c:1600]: Hm, fair point. I *think* we could in theory get it even with h2 and h3 so I'll change the call to streamclose... Thanks!

Comment (jay) [reply lib/http.c]: AFAICS streamclose calls [Curl_conncontrol](https://github.com/curl/curl/blob/curl-7_74_0/lib/connect.c#L1578) which doesn't actually signal to close the stream (if I'm reading it right), so I don't think it's going to stop that message

Comment (bagder) [reply lib/http.c]: the message `connection left intact` is still there and meant for the actual **connection**, so a stream getting closed early or not will not necessarily affect the connection the stream was sent over. So I think this is fine...

**Pull request closed without merging**